HarpGuru/Controls/ActivityLegendCell/analyseLegendKey/index.ts

```diff
@@ -0,0 +1,35 @@
+import { isPitchId, getPitchIds, DegreeIds } from 'harpstrata'
+
+import type { ActivityLegendCellProps } from '../types'
+
+type LegendKeyFacts = {
+  readonly isActive: boolean;
+  readonly counterpartDegreeId: DegreeIds;
+}
+
+const getIsActive = (props: ActivityLegendCellProps): boolean => {
+  const { itemId, activeIds } = props
+
+  const isActive = [...activeIds].includes(itemId)
+
+  return isActive
+}
+
+const getCounterpartDegreeId = (props: ActivityLegendCellProps): DegreeIds => {
+  const { activeHarpStrata: { rootPitchId }, itemId } = props
+
+  if (!isPitchId(itemId)) return itemId
+
+  const itemIndex = getPitchIds(rootPitchId).indexOf(itemId)
+
+  const { [itemIndex]: counterpartDegreeId } = Object.values(DegreeIds)
+
+  return counterpartDegreeId
+}
```

Comment:
I'm not too comfortable with this whole design but it might not be necessary for long if some design decisions I'm considering go through.